data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -62,7 +81,13 @@ private AggregateAction loadAggregateAction(final PluginFactory pluginFactory) {
 
         for (final Map.Entry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> groupEntry : groupsToConclude) {
             final Optional<Event> concludeGroupEvent = aggregateActionSynchronizer.concludeGroup(groupEntry.getKey(), groupEntry.getValue());
-            concludeGroupEvent.ifPresent(value -> recordsOut.add(new Record<>(value)));
+
+            if (concludeGroupEvent.isPresent()) {
+                recordsOut.add(new Record(concludeGroupEvent.get()));
+                actionConcludeGroupEventsForwardedCounter.increment();
```

Comment:
I don't know how micrometer is incrementing these counts. But, they must have some support for multiple threads. Do they give any guidance on how to use increment - whether it is better to accumulate as much as possible and then use `increment(double)` or if using `increment()` multiple times is just as fast?

Reply:
I couldn't find anything from Micrometer directly. But technically I guess it's probably better to have local variables and use `increment(double)` since I imagine under the hood something like an `AtomicDouble` is used? The performance improvement might be almost negligible though.

Reply:
We can also look into profiling it to see if it really matters anyway.

Reply:
It sounds like Micrometer's documentation doesn't call this out as a best practice. Since they don't, then it probably is not a major performance problem. So I don't see a need to make any changes right now. Thanks for checking their documentation!

Reply:
Haha you think we should revert the commit?

Reply:
I see your recent change. It's good. I just didn't want to slow down the PR if it wouldn't matter much.